File: source/CommaForNumber/comma-for-number.js
const TEXT = process.env.POPCLIP_TEXT;

function addOrRemoveCommaForNumber(number) {
    if (!number) throw new Error('no target number exists!');

    if (number.includes(",")) {
        return number.replace(/,/g, "");
    }

    var chunks = []
    while (number.length > 3) {
        chunks.push(number.slice(number.length-3));
        number = number.slice(0, number.length-3);
    }
    chunks.push(number);
    return chunks.reverse().join(",");
}

function main() {
    console.log(addOrRemoveCommaForNumber(TEXT));
}

main();

File: source/CommaForNumber.popclipextz/comma-for-number.py
import os


def remove_or_add_comma_for_number(number):
    """ remove or add for number string"""
    if ',' in number:
        return number.replace(',', '')

    chunks = []
    while len(number) > 3:
        chunks.append(number[-3:])
        number = number[:-3]
    chunks.append(number)

    return ','.join(reversed(chunks))


print(remove_or_add_comma_for_number(os.environ['POPCLIP_TEXT']), end="")
